src/sentry/static/sentry/app/components/search/sources/routeSource.tsx

```diff
@@ -0,0 +1,181 @@
+ */
+const mapFunc = (config: Config, context: Context | null = null) =>
+  (Array.isArray(config)
+    ? config
+    : context !== null
+    ? config(context)
+    : []
+  ).map(({items}) =>
+    items.filter(({show}) =>
+      typeof show === 'function' && context !== null ? show(context) : true
+    )
+  );
+
+type ItemProps = {
+  resultType: 'route';
+  sourceType: 'route';
+  to: string;
+};
+
+type Result = {
+  item: NavigationItem & ItemProps;
+  matches: FuseResultWithMatches<NavigationItem>['matches'];
+  score: number;
+};
+
+type RenderProps = {
+  isLoading: boolean;
+  /**
+   * Unused in this source
+   */
+  allResults: object[];
+  /**
+   * Matched results
+   */
+  results: Result[];
+};
+
+type DefaultProps = {
+  /**
+   * Fuse configuration for searching NavigationItem's
+   */
+  searchOptions: FuseOptions<NavigationItem>;
+};
+
+type Props = WithRouterProps &
+  DefaultProps & {
+    organization: Organization;
+    project: Project;
+    /**
+     * The string to search the navigation routes for
+     */
+    query: string;
+    /**
+     * Render function that renders the route matches
+     */
+    children: (props: RenderProps) => React.ReactNode;
+  };
+
+type State = {
+  /**
+   * A Fuse instance configured to search NavigationItem's
+   */
+  fuzzy: null | Fuse<NavigationItem, FuseOptions<NavigationItem>>;
+};
+
+class RouteSource extends React.Component<Props, State> {
+  static defaultProps: DefaultProps = {
+    searchOptions: {},
+  };
+
+  constructor(props: Props) {
+    super(props);
+
+    this.createSearch();
+  }
+
+  state: State = {
+    fuzzy: null,
+  };
+
+  componentDidUpdate(prevProps: Props) {
+    if (
+      prevProps.project === this.props.project &&
+      prevProps.organization === this.props.organization
+    ) {
+      return;
+    }
+
+    this.createSearch();
+  }
+
+  async createSearch() {
+    const {project, organization} = this.props;
+
+    const context = {
+      project,
+      organization,
+      access: new Set(organization?.access ?? []),
+      features: new Set(project?.features ?? []),
+    } as Context;
+
+    const searchMap = flattenDepth<NavigationItem>(
+      [
+        mapFunc(accountSettingsNavigation),
+        mapFunc(projectSettingsNavigation, context),
+        mapFunc(organizationSettingsNavigation, context),
+      ],
+      2
+    );
+
+    const options = {
+      ...this.props.searchOptions,
+      keys: ['title', 'description'],
+    };
+
+    const fuzzy = await createFuzzySearch<NavigationItem>(searchMap ?? [], options);
+    this.setState({fuzzy});
+  }
+
+  render() {
+    const {query, params, children} = this.props;
+
+    const results =
+      this.state.fuzzy?.search<NavigationItem, true, true>(query).map(
+        ({item, ...rest}) =>
+          ({
+            item: {
+              ...item,
+              sourceType: 'route',
+              resultType: 'route',
+              to: `${replaceRouterParams(item.path, params)}`,
```

Comment:
this function should always return a string

```suggestion
              to: replaceRouterParams(item.path, params),
```

Reply:
I'll just leave it as is for now, just so I don't change too many things in this PR

Reply:
are you sure? it's only one line 🙂

Reply:
I'll make a sep PR :)